src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -563,21 +607,31 @@ void Snippet::SnippetJitExecutor::schedule_6d(const std::vector<MemoryPtr>& inMe
     const auto& dom = parallel_exec_domain;
     // < N, C, H, W > < 1, 1, N, C*H*W>
     const auto& callable = schedule.get_callable<kernel>();
-#if defined(__linux__) && defined(SNIPPETS_DEBUG_CAPS)
+#if defined(__linux__) && defined(OPENVINO_ARCH_X86_64) && defined(SNIPPETS_DEBUG_CAPS)
     segfault_detector();
 #endif
-    parallel_for5d(dom[0], dom[1], dom[2], dom[3], dom[4],
-        [&](int64_t d0, int64_t d1, int64_t d2, int64_t d3, int64_t d4) {
-            int64_t indexes[] = {d0, d1, d2, d3, d4};
-            jit_snippets_call_args call_args;
-            update_ptrs(call_args, inMemPtrs, outMemPtrs);
+    parallel_nt(0, [&](const int ithr, const int nthr) {
+        jit_snippets_call_args call_args;
+        update_ptrs(call_args, inMemPtrs, outMemPtrs);
+
+        size_t start = 0, end = 0;
+        splitter(harnessWorkAmount, nthr, ithr, start, end);
+
+        int64_t indexes[] = {0, 0, 0, 0, 0};
+        for (size_t iwork = start; iwork < end; ++iwork) {
+            size_t tmp = iwork;
+            for (int j = 4; j >= 0; j--) {
+                indexes[j] = static_cast<int64_t>(tmp % dom[j]);
+                tmp /= dom[j];
+            }
             callable(&call_args, indexes);
-        });
+        }
```

Comment:
This implementation is the same as `schedule_nt` impl and as slow as `schedule_nt` impl below.
You can reuse here impl of [for_5d](https://github.com/openvinotoolkit/openvino/blob/a10252019212ab95909e5cd881e1488286bfbbce/src/core/include/openvino/core/parallel.hpp#L625) but init ptrs once.
In the https://github.com/openvinotoolkit/openvino/pull/21922 I already implemented the same way for dynamic case:
https://github.com/openvinotoolkit/openvino/blob/0b93adc172809c4fa2ea70b2a4e2433eb6d7bc77/src/plugins/intel_cpu/src/nodes/subgraph.cpp#L877-L903

I think you can do the same:
```
parallel_nt(0, [&](const int ithr, const int nthr) {
        jit_snippets_call_args call_args;
        update_ptrs(call_args, inMemPtrs, outMemPtrs);

        size_t start = 0, end = 0;
        splitter(harnessWorkAmount, nthr, ithr, start, end);

        int64_t indexes[] = {0, 0, 0, 0, 0};
        parallel_it_init(start, indexes[0], dom[0], indexes[1], dom[1], indexes[2], dom[2], indexes[3], dom[3], indexes[4], dom[4]);  // <- probably you init the separate instances for dimensions to avoid accessing to vector elements during execution
        for (size_t iwork = start; iwork < end; ++iwork) {
            callable(&call_args, indexes);
            parallel_it_step(indexes[0], dom[0], indexes[1], dom[1], indexes[2], dom[2], indexes[3], dom[3], indexes[4], dom[4]);
        }
    });

```

What do you think?

Reply:
Cool! Applied. Thanks Alexandra!